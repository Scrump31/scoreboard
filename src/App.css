/* Body */
/* Scoreboard */
.scoreboard-app {
  margin-top: 1rem;
  background-color: black;
  color: white;
  border-radius: 1rem;
  width: 700px;
}
.scoreboard-title {
  text-align: center;
  font-size: 3.5rem;
}
@media screen and (max-width: 812px) and (orientation: landscape) {
  .scoreboard-title {
    text-align: right;
  }
 }

/* Players */
.players-container {
  border-top: 2px solid white;
  background-color: #598aea;
  font-size: 3rem;
}
.player-score {
  display: inline-block;
  width: 1.8em;
  text-align: center;
}

.player-name+.player-counter {
  padding: 5px;
}
.player-counter {
  text-align: right;
}
.player-name:hover {
  text-decoration: line-through;
  text-decoration-color: #d9534f;
  -moz-text-decoration-color: #d9534f;
}
.player-name:hover .glyphicon-remove {
  visibility: visible;
}

@media only screen and (max-width: 991px) {
    /* For mobile devices: */
    .player-name .glyphicon-remove {
      visibility: visible;
    }
  }
 .glyphicon-remove {
  visibility: hidden;
  padding-left: 5px;
  font-size: 0.9em;
  color: #d9534f;
  border: none;
  background: none;
}

.horizonal-rule {
  border: 1px solid white;
}
.btn {
  box-shadow: 0 1px 6px 1px black;
}
/* Decrement Button */
.btn-danger {
   width: 4.6rem;
}

/* Player Transitions */
.fade-enter {
  opacity: 0;
}

/* end state of animation */
.fade-enter-active {
  opacity: 1;
  transition: 0.5s ease-in all;
}

.fade-leave {
  opacity: 1
}

.fade-leave-active {
  opacity: 0;
  transition: 1s ease-out all;
}


/* Stats Table */
.stats {
  font-size: 2rem;
  margin-bottom: 0;
}
.table>tbody>.total-players-td-row>td {
  border:none;
}

/* Add Player */
form {
  margin-top: 1.5%;
}

/* Timer */
.progress {
  margin-bottom: 1rem;
  border: 1px solid #fff;
}
.progress span {
  background: #000;

  height: 100%;
  float: right;
}
.time {
  font-size: 2rem;
}
.controls {
  padding-bottom: 1rem;
}
.pause {
  margin: 0 1rem;
}
