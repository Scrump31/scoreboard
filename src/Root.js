import React from 'react';
import { createStore } from 'redux';
import { Provider } from 'react-redux';

import store from './reducers/index';

export default ({ children }) => (
  <Provider store={createStore(store)}>{children}</Provider>
);
