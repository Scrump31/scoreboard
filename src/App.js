// import React, { Component } from 'react';
import React, { Component } from 'react';
import { Grid, Row, Col } from 'react-bootstrap';
import Stats from './components/Stats';
import Timer from './components/Timer';
import Player from './components/Player';

import './App.css';

export default class App extends Component {
  constructor(props) {
    super(props);
    this.state = { players: this.props.initialPlayers, score: 0 };
  }
  // getInitialState() {
  //
  // }

  render() {
    return (
      <Grid className="scoreboard-app">
        <Row>
          <Col md={4} className="text-left stats">
            <Stats
              totalPlayers={this.state.players.length}
              totalPoints={0}
            />
          </Col>
          <Col md={5} className="scoreboard-title">
            {this.props.scoreboard}
          </Col>
          <Col md={3} className="text-right timer"><Timer /></Col>
        </Row>

        <Row className="players-container">
          {this.state.players.map(player =>
            <Player
              name={player.name}
              score={player.score}
              key={player.id}
            />)}
        </Row>
      </Grid>
    );
  }

}
App.propTypes = {
  scoreboard: React.PropTypes.string,
  initialPlayers: React.PropTypes.arrayOf(React.PropTypes.shape({
    name: React.PropTypes.string,
    score: React.PropTypes.number,
  })).isRequired,
};
App.defaultProps = {
  scoreboard: 'Scoreboard',
};
