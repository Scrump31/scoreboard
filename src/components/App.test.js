import React from 'react';
import { mount } from 'enzyme';
import Root from '../Root';
import App from './App';

describe('<App />', () => {
  it('renders without crashing', () => {
    const component = mount(
      <Root>
        <App />
      </Root>,
    );
    expect(component).toMatchSnapshot();
    component.unmount();
  });
});
