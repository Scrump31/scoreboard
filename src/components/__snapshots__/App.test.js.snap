// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
<_default>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <App>
      <Grid
        bsClass="container"
        className="scoreboard-app"
        componentClass="div"
        fluid={false}
      >
        <div
          className="scoreboard-app container"
        >
          <Row
            bsClass="row"
            componentClass="div"
          >
            <div
              className="row"
            >
              <Col
                bsClass="col"
                className="text-left stats"
                componentClass="div"
                sm={4}
                xs={6}
              >
                <div
                  className="text-left stats col-sm-4 col-xs-6"
                >
                  <Connect(Stats)>
                    <Stats
                      dispatch={[Function]}
                      players={
                        Array [
                          Object {
                            "id": "1",
                            "name": "Player One",
                            "score": 31,
                          },
                          Object {
                            "id": "2",
                            "name": "Player Two",
                            "score": 35,
                          },
                          Object {
                            "id": "3",
                            "name": "Player Three",
                            "score": 42,
                          },
                        ]
                      }
                    >
                      <Table
                        bordered={false}
                        bsClass="table"
                        className="stats"
                        condensed={false}
                        hover={false}
                        responsive={false}
                        striped={false}
                      >
                        <table
                          className="stats table"
                        >
                          <tbody>
                            <tr
                              className="total-players-td-row"
                            >
                              <td>
                                <strong>
                                  Players:
                                </strong>
                              </td>
                              <td
                                id="total_players"
                              >
                                3
                              </td>
                            </tr>
                            <tr>
                              <td>
                                <strong>
                                  Total Points:
                                </strong>
                              </td>
                              <td
                                id="total_score"
                              >
                                108
                              </td>
                            </tr>
                          </tbody>
                        </table>
                      </Table>
                    </Stats>
                  </Connect(Stats)>
                </div>
              </Col>
              <Col
                bsClass="col"
                className="scoreboard-title"
                componentClass="div"
                sm={4}
                xs={6}
              >
                <div
                  className="scoreboard-title col-sm-4 col-xs-6"
                >
                  Scoreboard
                </div>
              </Col>
              <Col
                bsClass="col"
                className="text-center timer"
                componentClass="div"
                sm={4}
                xs={12}
              >
                <div
                  className="text-center timer col-sm-4 col-xs-12"
                >
                  <BasicTimrExample>
                    <div>
                      <div
                        className="time"
                      >
                        1:00
                      </div>
                      <div
                        className="progress"
                      >
                        <span
                          style={
                            Object {
                              "width": "0%",
                            }
                          }
                        />
                      </div>
                      <div
                        className="controls"
                      >
                        <button
                          className="start btn btn-sm btn-success"
                          onClick={[Function]}
                        >
                          Start
                        </button>
                        <button
                          className="pause btn btn-sm btn-default"
                          onClick={[Function]}
                        >
                          Pause
                        </button>
                        <button
                          className="stop btn btn-sm btn-danger"
                          onClick={[Function]}
                        >
                          Stop
                        </button>
                      </div>
                    </div>
                  </BasicTimrExample>
                </div>
              </Col>
            </div>
          </Row>
          <Row
            bsClass="row"
            className="players-container"
            componentClass="div"
          >
            <div
              className="players-container row"
            >
              <Col
                bsClass="col"
                componentClass="div"
                xs={12}
              >
                <div
                  className="col-xs-12"
                >
                  <Connect(PlayerList)>
                    <PlayerList
                      deletePlayer={[Function]}
                      players={
                        Array [
                          Object {
                            "id": "1",
                            "name": "Player One",
                            "score": 31,
                          },
                          Object {
                            "id": "2",
                            "name": "Player Two",
                            "score": 35,
                          },
                          Object {
                            "id": "3",
                            "name": "Player Three",
                            "score": 42,
                          },
                        ]
                      }
                      updateScore={[Function]}
                    >
                      <TransitionGroup
                        childFactory={[Function]}
                        component="div"
                      >
                        <div>
                          <CSSTransition
                            classNames="fade"
                            in={true}
                            key=".$1"
                            onExited={[Function]}
                            timeout={500}
                          >
                            <Transition
                              appear={false}
                              enter={true}
                              exit={true}
                              in={true}
                              mountOnEnter={false}
                              onEnter={[Function]}
                              onEntered={[Function]}
                              onEntering={[Function]}
                              onExit={[Function]}
                              onExited={[Function]}
                              onExiting={[Function]}
                              timeout={500}
                              unmountOnExit={false}
                            >
                              <Player
                                deletePlayer={[Function]}
                                id="1"
                                key="1"
                                name="Player One"
                                score={31}
                                updateScore={[Function]}
                              >
                                <Row
                                  bsClass="row"
                                  componentClass="div"
                                >
                                  <div
                                    className="row"
                                  >
                                    <Col
                                      bsClass="col"
                                      className="player-name"
                                      componentClass="div"
                                      xs={9}
                                    >
                                      <div
                                        className="player-name col-xs-9"
                                      >
                                        <div>
                                          Player One
                                          <button
                                            className="glyphicon glyphicon-remove"
                                            onClick={[Function]}
                                          />
                                        </div>
                                      </div>
                                    </Col>
                                    <Col
                                      bsClass="col"
                                      className="player-counter"
                                      componentClass="div"
                                      xs={3}
                                    >
                                      <div
                                        className="player-counter col-xs-3"
                                      >
                                        <Counter
                                          id="1"
                                          score={31}
                                          updateScore={[Function]}
                                        >
                                          <div>
                                            <Button
                                              active={false}
                                              block={false}
                                              bsClass="btn"
                                              bsSize="large"
                                              bsStyle="danger"
                                              disabled={false}
                                              onClick={[Function]}
                                            >
                                              <button
                                                className="btn btn-lg btn-danger"
                                                disabled={false}
                                                onClick={[Function]}
                                                type="button"
                                              >
                                                -
                                              </button>
                                            </Button>
                                            <span
                                              className="player-score"
                                            >
                                              31
                                            </span>
                                            <Button
                                              active={false}
                                              block={false}
                                              bsClass="btn"
                                              bsSize="large"
                                              bsStyle="success"
                                              disabled={false}
                                              onClick={[Function]}
                                            >
                                              <button
                                                className="btn btn-lg btn-success"
                                                disabled={false}
                                                onClick={[Function]}
                                                type="button"
                                              >
                                                +
                                              </button>
                                            </Button>
                                          </div>
                                        </Counter>
                                      </div>
                                    </Col>
                                    <Col
                                      bsClass="col"
                                      className="horizonal-rule"
                                      componentClass="div"
                                      md={12}
                                    >
                                      <div
                                        className="horizonal-rule col-md-12"
                                      />
                                    </Col>
                                  </div>
                                </Row>
                              </Player>
                            </Transition>
                          </CSSTransition>
                          <CSSTransition
                            classNames="fade"
                            in={true}
                            key=".$2"
                            onExited={[Function]}
                            timeout={500}
                          >
                            <Transition
                              appear={false}
                              enter={true}
                              exit={true}
                              in={true}
                              mountOnEnter={false}
                              onEnter={[Function]}
                              onEntered={[Function]}
                              onEntering={[Function]}
                              onExit={[Function]}
                              onExited={[Function]}
                              onExiting={[Function]}
                              timeout={500}
                              unmountOnExit={false}
                            >
                              <Player
                                deletePlayer={[Function]}
                                id="2"
                                key="2"
                                name="Player Two"
                                score={35}
                                updateScore={[Function]}
                              >
                                <Row
                                  bsClass="row"
                                  componentClass="div"
                                >
                                  <div
                                    className="row"
                                  >
                                    <Col
                                      bsClass="col"
                                      className="player-name"
                                      componentClass="div"
                                      xs={9}
                                    >
                                      <div
                                        className="player-name col-xs-9"
                                      >
                                        <div>
                                          Player Two
                                          <button
                                            className="glyphicon glyphicon-remove"
                                            onClick={[Function]}
                                          />
                                        </div>
                                      </div>
                                    </Col>
                                    <Col
                                      bsClass="col"
                                      className="player-counter"
                                      componentClass="div"
                                      xs={3}
                                    >
                                      <div
                                        className="player-counter col-xs-3"
                                      >
                                        <Counter
                                          id="2"
                                          score={35}
                                          updateScore={[Function]}
                                        >
                                          <div>
                                            <Button
                                              active={false}
                                              block={false}
                                              bsClass="btn"
                                              bsSize="large"
                                              bsStyle="danger"
                                              disabled={false}
                                              onClick={[Function]}
                                            >
                                              <button
                                                className="btn btn-lg btn-danger"
                                                disabled={false}
                                                onClick={[Function]}
                                                type="button"
                                              >
                                                -
                                              </button>
                                            </Button>
                                            <span
                                              className="player-score"
                                            >
                                              35
                                            </span>
                                            <Button
                                              active={false}
                                              block={false}
                                              bsClass="btn"
                                              bsSize="large"
                                              bsStyle="success"
                                              disabled={false}
                                              onClick={[Function]}
                                            >
                                              <button
                                                className="btn btn-lg btn-success"
                                                disabled={false}
                                                onClick={[Function]}
                                                type="button"
                                              >
                                                +
                                              </button>
                                            </Button>
                                          </div>
                                        </Counter>
                                      </div>
                                    </Col>
                                    <Col
                                      bsClass="col"
                                      className="horizonal-rule"
                                      componentClass="div"
                                      md={12}
                                    >
                                      <div
                                        className="horizonal-rule col-md-12"
                                      />
                                    </Col>
                                  </div>
                                </Row>
                              </Player>
                            </Transition>
                          </CSSTransition>
                          <CSSTransition
                            classNames="fade"
                            in={true}
                            key=".$3"
                            onExited={[Function]}
                            timeout={500}
                          >
                            <Transition
                              appear={false}
                              enter={true}
                              exit={true}
                              in={true}
                              mountOnEnter={false}
                              onEnter={[Function]}
                              onEntered={[Function]}
                              onEntering={[Function]}
                              onExit={[Function]}
                              onExited={[Function]}
                              onExiting={[Function]}
                              timeout={500}
                              unmountOnExit={false}
                            >
                              <Player
                                deletePlayer={[Function]}
                                id="3"
                                key="3"
                                name="Player Three"
                                score={42}
                                updateScore={[Function]}
                              >
                                <Row
                                  bsClass="row"
                                  componentClass="div"
                                >
                                  <div
                                    className="row"
                                  >
                                    <Col
                                      bsClass="col"
                                      className="player-name"
                                      componentClass="div"
                                      xs={9}
                                    >
                                      <div
                                        className="player-name col-xs-9"
                                      >
                                        <div>
                                          Player Three
                                          <button
                                            className="glyphicon glyphicon-remove"
                                            onClick={[Function]}
                                          />
                                        </div>
                                      </div>
                                    </Col>
                                    <Col
                                      bsClass="col"
                                      className="player-counter"
                                      componentClass="div"
                                      xs={3}
                                    >
                                      <div
                                        className="player-counter col-xs-3"
                                      >
                                        <Counter
                                          id="3"
                                          score={42}
                                          updateScore={[Function]}
                                        >
                                          <div>
                                            <Button
                                              active={false}
                                              block={false}
                                              bsClass="btn"
                                              bsSize="large"
                                              bsStyle="danger"
                                              disabled={false}
                                              onClick={[Function]}
                                            >
                                              <button
                                                className="btn btn-lg btn-danger"
                                                disabled={false}
                                                onClick={[Function]}
                                                type="button"
                                              >
                                                -
                                              </button>
                                            </Button>
                                            <span
                                              className="player-score"
                                            >
                                              42
                                            </span>
                                            <Button
                                              active={false}
                                              block={false}
                                              bsClass="btn"
                                              bsSize="large"
                                              bsStyle="success"
                                              disabled={false}
                                              onClick={[Function]}
                                            >
                                              <button
                                                className="btn btn-lg btn-success"
                                                disabled={false}
                                                onClick={[Function]}
                                                type="button"
                                              >
                                                +
                                              </button>
                                            </Button>
                                          </div>
                                        </Counter>
                                      </div>
                                    </Col>
                                    <Col
                                      bsClass="col"
                                      className="horizonal-rule"
                                      componentClass="div"
                                      md={12}
                                    >
                                      <div
                                        className="horizonal-rule col-md-12"
                                      />
                                    </Col>
                                  </div>
                                </Row>
                              </Player>
                            </Transition>
                          </CSSTransition>
                        </div>
                      </TransitionGroup>
                    </PlayerList>
                  </Connect(PlayerList)>
                </div>
              </Col>
            </div>
          </Row>
          <Row
            bsClass="row"
            componentClass="div"
          >
            <div
              className="row"
            >
              <Col
                bsClass="col"
                componentClass="div"
                xs={12}
              >
                <div
                  className="col-xs-12"
                >
                  <Connect(AddPlayer)>
                    <AddPlayer
                      createPlayer={[Function]}
                    >
                      <form
                        onSubmit={[Function]}
                      >
                        <FormGroup
                          bsClass="form-group"
                        >
                          <div
                            className="form-group"
                          >
                            <InputGroup
                              bsClass="input-group"
                            >
                              <span
                                className="input-group"
                              >
                                <FormControl
                                  bsClass="form-control"
                                  componentClass="input"
                                  onChange={[Function]}
                                  placeholder="Add a new player..."
                                  type="text"
                                  value=""
                                >
                                  <input
                                    className="form-control"
                                    onChange={[Function]}
                                    placeholder="Add a new player..."
                                    type="text"
                                    value=""
                                  />
                                </FormControl>
                                <InputGroupButton
                                  bsClass="input-group-btn"
                                >
                                  <span
                                    className="input-group-btn"
                                  >
                                    <Button
                                      active={false}
                                      block={false}
                                      bsClass="btn"
                                      bsStyle="default"
                                      disabled={false}
                                      type="submit"
                                    >
                                      <button
                                        className="btn btn-default"
                                        disabled={false}
                                        type="submit"
                                      >
                                        Add
                                      </button>
                                    </Button>
                                  </span>
                                </InputGroupButton>
                              </span>
                            </InputGroup>
                          </div>
                        </FormGroup>
                      </form>
                    </AddPlayer>
                  </Connect(AddPlayer)>
                </div>
              </Col>
            </div>
          </Row>
        </div>
      </Grid>
    </App>
  </Provider>
</_default>
`;
